### PR TITLE
#202 #201

### PR DESCRIPTION
**Pull request by banasiakadam60**
Allow user to define task name + show computation cost and time

### CODE REVIEW

**Comment by banasiakadam60** [comment]
@badb Can it be merged? I've moved all commits related with #201 and #202 from 'names' branch to this one to clean up this mess I made. I've joined this two issues becouse both need a change in AppMainWindow.ui, so I could pushed it as one PR, or two related PRs. IMO one PR is better option.

**Comment by badb** [comment]
It have to wait with being merge - there's not time to add it to relase 0.3 version (because there is no time to test it and still a lot of debugging with other problems. 
One new PR is ok imho. Can you add some tests to pass codecdov patch? 
And one small issue: default lux render task is loaded properly after clicking "reset to defaults", but not after changing task type combo box.

**Comment by banasiakadam60** [comment]
OK. Oh, thanks, I have somehow missed this issue :)

File: golem/transactions/ethereum/ethereumtransactionsystem.py [hunk 1]
```diff
@@ -59,3 +60,15 @@ def get_incoming_payments(self):
...
+        """
+        Get cost of subtasks
+        :param subtasks: List of subtasks ID
+        :return: Cost of subtasks
+        """
+        cost = 0.0
+        if self.payments_keeper:
+            for subtask in subtasks:
+                cost += self.payments_keeper.get_payment(subtask)
+        return cost / denoms.ether
```
Comment (badb) [inline golem/transactions/ethereum/ethereumtransactionsystem.py]: One small thing: in other places where ether division is implemented there is also imported float division from python 3. Can you add from **future** import division here?

Comment (banasiakadam60) [inline golem/transactions/ethereum/ethereumtransactionsystem.py]: OK, done :)

File: golem/transactions/ethereum/ethereumtransactionsystem.py [hunk 2]
```diff
@@ -59,3 +59,15 @@ def get_incoming_payments(self):
...
+    def get_payment_for_subtasks(self, subtasks):
+        """
+        Get cost of subtasks
+        :param subtasks: List of subtasks ID
+        :return: Cost of subtasks
+        """
+        cost = 0
+        if self.payments_keeper:
+            for subtask in subtasks:
+                cost += self.payments_keeper.get_payment(subtask)
```
Comment (mfranciszkiewicz) [inline golem/transactions/ethereum/ethereumtransactionsystem.py]: Wouldn't it be better to select multiple `Payment.cost`s in a single query? If you're concerned with query size, you can:
- split the query into batches
- later on, when a task <- subtask mapping is stored in the db, query with `task_id` (hopefully soon)

Plus, how about storing a (lazily initialized) sum and updating it on each payment?

Comment (mfranciszkiewicz) [reply golem/transactions/ethereum/ethereumtransactionsystem.py]: On second thought: I think we should postpone the optimizations and merge this PR.

**Pull request merged**